lib/index.coffee

```diff
@@ -0,0 +1,134 @@
+  if thrift_options.timeout?
+    debug "adding timeout listener"
+    connection.on "timeout", ->
+      debug "in timeout callback"
+      connection.__ended = true
+
+
+# create_pool initializes a generic-pool
+#   @param thrift library to use to in create_cb
+#   @param pool_options, host/port are used in create_cb
+#          max, min, idleTimeouts are used by generic pool
+#   @param thrift_options used in create_cb
+#
+create_pool = (thrift, pool_options = {}, thrift_options = {}) ->
+  pool = genericPool.Pool
+    name: "thrift"
+    create: (cb) ->
+      create_cb thrift, pool_options, thrift_options, cb
+    destroy: (connection) ->
+      debug "in destroy"
+      connection.end()
+    validate: (connection) ->
+      debug "in validate"
+      not connection.__ended
+    max: pool_options.max_connections
+    min: pool_options.min_connections
+    idleTimeoutMillis: pool_options.idle_timeout
+
+module.exports = (thrift, service, pool_options = {}, thrift_options = {}) ->
+
+  throw new Error "You must specify #{key}" for key in ["host", "port"] when not pool_options[key]
+
+  pool_options = _(pool_options).defaults
+    max_connections: 2 # Max number of connections to keep open
+    min_connections: 0 # Min number of connections to keep open
+    idle_timeout: 30000 # Time (in ms) to wait until closing idle connections
+
+  pool = create_pool thrift, pool_options, thrift_options
+
+  # add_listeners adds listeners for error, close, and timeout
+  #   @param connection, connection to add listeners to
+  #   @param cb_error, callback to attach to "error" listener
+  #   @param cb_timeout, callback to attach to "timeout" listener
+  #   @param cb_close, callback to attach to "close" listener
+  add_listeners = (connection, cb_error, cb_timeout, cb_close) ->
+    connection.on "error", cb_error
+    connection.on "close", cb_close
+    if thrift_options.timeout?
+      connection.on "timeout", cb_timeout
+
+  # remove_listeners removes error, timeout, and close listeners with given callbacks
+  #   @param connection, connection to remove listeners from
+  #   @param cb_error, error callback to remove from "error" listener
+  #   @param cb_timeout, timeout callback to remove from "timeout" listener
+  #   @param cb_close, close callback to remove from "close" listener
+  remove_listeners = (connection, cb_error, cb_timeout, cb_close) ->
+    connection.removeListener "error", cb_error
+    connection.removeListener "close", cb_close
+    if thrift_options.timeout?
+      connection.removeListener "timeout", cb_timeout
+
+  # wrap_thrift_fn when called with a function and arguments/callback:
+  #   - acquires a connection
+  #   - adds additional connection event listeners
+  #   - creates a client with the acquired connection
+  #   - calls client with fn and passed args and callback
+  #   - connection is released before results are returned
+  #  @return, function that takes in arguments and a callback
+  #
```

Comment:
super ocd nit: sometimes we have an empty comment at the end and sometimes we don't. I slightly prefer not having one, but I definitely like being consistent